pkg/virtctl/vmexport/vmexport.go

```diff
@@ -0,0 +1,440 @@
+	"kubevirt.io/kubevirt/pkg/util"
+	"kubevirt.io/kubevirt/pkg/virtctl/templates"
+)
+
+const (
+	// processingWaitInterval is the time interval used to wait for a virtualMachineExport to be ready
+	processingWaitInterval = 2 * time.Second
+	// processingWaitTotal is the maximum time used to wait for a virtualMachineExport to be ready
+	processingWaitTotal = 24 * time.Hour
+
+	// ErrRequiredFlag serves as error message when a mandatory flag is missing
+	ErrRequiredFlag = "Need to specify the '%s' flag when using '%s'"
+	// ErrIncompatibleFlag serves as error message when an incompatible flag is used
+	ErrIncompatibleFlag = "The '%s' flag is incompatible with '%s'"
+	// ErrRequiredExportType serves as error message when no export kind is provided
+	ErrRequiredExportType = "Need to specify export kind when attempting to create a VirtualMachineExport [pvc|vm|snapshot]"
+	// ErrIncompatibleExportType serves as error message when an export kind is provided with an incompatible argument
+	ErrIncompatibleExportType = "Should not specify export kind"
+	// ErrBadExportArguments serves as error message when vmexport is used with bad arguments
+	ErrBadExportArguments = "Expecting two args: vmexport function [create|delete|download] and VirtualMachineExport name"
+)
+
+var (
+	// Flags
+	vm           string
+	snapshot     string
+	pvc          string
+	outputFile   string
+	shouldCreate bool
+	volumeName   string
+
+	// vmexport info
+	vmexportName string
+	resourceName string
+	resourceKind string
+	funcName     string
+)
+
+type exportFunc func(client kubecli.KubevirtClient, namespace string) error
+
+type command struct {
+	clientConfig clientcmd.ClientConfig
+}
+
+var exportFunction exportFunc
+
+// usage provides several valid usage examples of vmexport
+func usage() string {
+	usage := `# Create a VirtualMachineExport to export a volume from a virtual machine:
+	{{ProgramName}} vmexport create vm1-export --vm=vm1
+  
+	# Create a VirtualMachineExport to export a volume from a virtual machine snapshot
+	{{ProgramName}} vmexport create snap1-export --snapshot=snap1
+  
+	# Create a VirtualMachineExport to export a volume from a PVC
+	{{ProgramName}} vmexport create pvc1-export --pvc=pvc1
+  
+	# Delete a VirtualMachineExport resource
+	{{ProgramName}} vmexport delete snap1-export
+  
+	# Download a volume from an already existing VirtualMachineExport
+	{{ProgramName}} vmexport download vm1-export --volume=volume1 --output=disk.img.gz
+  
+	# Create a VirtualMachineExport and download the requested volume from it
+	{{ProgramName}} vmexport download vm1-export --create --volume=volume1 --output=disk.img.gz`
+
+	return usage
+}
+
+// NewVirtualMachineExportCommand returns a cobra.Command to handle the export process
+func NewVirtualMachineExportCommand(clientConfig clientcmd.ClientConfig) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:     "vmexport",
+		Short:   "Export a VM volume.",
+		Example: usage(),
+		Args:    cobra.MaximumNArgs(2),
```

Comment:
you can use instead: templates.ExactArgs("memory-dump", 2)
and then you dont need to check `if len(args) != 2 {`

Reply:
I'll check it, thanks!